include/deal.II/sundials/ida_interface.h

```diff
@@ -0,0 +1,476 @@
+     * the paramaters are parsed from a file, the internal parameters are automatically
+     * updated.
+     *
+     * The following parameters are declared:
+     *
+     * @code
+     * set Absolute error tolerance                      = 0.000001
+     * set Final time                                    = 1.000000
+     * set Ignore algebraic terms for error computations = true
+     * set Initial condition Newton max iterations       = 5
+     * set Initial condition Newton parameter            = 0.330000
+     * set Initial condition type                        = use_y_diff
+     * set Initial condition type after restart          = use_y_dot
+     * set Initial step size                             = 0.000100
+     * set Initial time                                  = 0.000000
+     * set Maximum number of nonlinear iterations        = 10
+     * set Maximum order of BDF                          = 5
+     * set Min step size                                 = 0.000001
+     * set Relative error tolerance                      = 0.000010
+     * set Show output of time steps                     = true
+     * set Time units between each output                = 0.05
+     * set Use local tolerances                          = false
+     * @endcode
+     *
+     * These are one-to-one with the options you can pass at construction time.
+     *
+     * Those options are set as default value in the ParameterHandler object
+     * `prm`.
+     */
+    virtual void add_parameters(ParameterHandler &prm);
+
+    /**
+     * Evolve. This function returns the final number of steps.
+     */
+    unsigned int solve_dae(VectorType &solution,
+                           VectorType &solution_dot);
+
+    /**
+     * Clear internal memory, and start with clean objects. This function is
+     * called when the simulation start and when the user return true to a call
+     * to solver_should_restart()
+     */
+    void reset_dae(const double t,
+                   VectorType &y,
+                   VectorType &yp,
+                   double h,
+                   bool first_step);
+
+    /**
+     * Return a newly allocated shared_ptr<VectorType>.
+     */
+    std::function<std::shared_ptr<VectorType>()> create_new_vector;
+
+    /**
+     * Compute residual.
+     */
+    std::function<int(const double t,
+                      const VectorType &y,
+                      const VectorType &y_dot,
+                      VectorType &res)> residual;
+
+    /**
+     * Compute Jacobian.
+     */
+    std::function<int(const double t,
+                      const VectorType &y,
+                      const VectorType &y_dot,
+                      const double alpha)> setup_jacobian;
+
+    /**
+     * Solve linear system.
+     */
+    std::function<int(const VectorType &rhs, VectorType &dst)> solve_jacobian_system;
+
+    /**
+     * Store solutions to file.
+     */
+    std::function<void (const double t,
+                        const VectorType &sol,
```

Comment:
I think that's the wrong documentation -- a user can attach any kind of function here. The function is probably called every time step, right? If so, we use a `boost::signals2::signal` for this in other places.

Reply:
Not at every time step, but at fixed times. I'll document this better.

Reply:
The point about the signal is that, just like for linear solvers, one may want to attach one or more listeners to this signal. That's possible with a signal, but not with a `std::function`.